apps/frontend/src/modules/chat/slices/chat.slice.ts

```diff
@@ -0,0 +1,136 @@
+import { createSlice, type PayloadAction } from "@reduxjs/toolkit";
+
+import { DataStatus } from "~/libs/enums/enums.js";
+import { type ValueOf } from "~/libs/types/types.js";
+import { type SelectedCategory } from "~/modules/categories/categories.js";
+import { ChatMessageAuthor, ChatMessageType } from "~/modules/chat/chat.js";
+import { buttonsModeOption } from "~/pages/chat/libs/enums/enums.js";
+
+import {
+	type ChatMessageDto,
+	type TaskCreateDto,
+	type TaskMessage,
+} from "../libs/types/types.js";
+import {
+	createTasksFromSuggestions,
+	getTasksForCategories,
+	initConversation,
+} from "./actions.js";
+
+type State = {
+	buttonsMode: ValueOf<typeof buttonsModeOption>;
+	dataStatus: ValueOf<typeof DataStatus>;
+	messages: Omit<ChatMessageDto, "createdAt" | "id">[];
+	selectedCategories: SelectedCategory[];
+	taskSuggestions: TaskCreateDto[];
+	threadId: null | string;
+};
+
+const initialState: State = {
+	buttonsMode: buttonsModeOption.SUGGESTIONS_CREATION,
+	dataStatus: DataStatus.IDLE,
+	messages: [],
+	selectedCategories: [],
+	taskSuggestions: [],
+	threadId: null,
+};
+
+const { actions, name, reducer } = createSlice({
+	extraReducers(builder) {
+		builder
+			.addCase(createTasksFromSuggestions.pending, (state) => {
+				state.taskSuggestions = [];
+				state.dataStatus = DataStatus.PENDING;
+			})
+			.addCase(createTasksFromSuggestions.fulfilled, (state) => {
+				state.dataStatus = DataStatus.FULFILLED;
+			})
+			.addCase(createTasksFromSuggestions.rejected, (state) => {
+				state.dataStatus = DataStatus.FULFILLED;
+			})
+
+			.addCase(initConversation.pending, (state) => {
+				state.dataStatus = DataStatus.PENDING;
+			})
+			.addCase(initConversation.fulfilled, (state, action) => {
+				state.threadId = action.payload.threadId;
+
+				state.dataStatus = DataStatus.FULFILLED;
+			})
+			.addCase(initConversation.rejected, (state) => {
+				state.dataStatus = DataStatus.REJECTED;
+			})
+
+			.addCase(getTasksForCategories.pending, (state) => {
+				state.dataStatus = DataStatus.PENDING;
+			})
+			.addCase(getTasksForCategories.fulfilled, (state, action) => {
+				state.dataStatus = DataStatus.FULFILLED;
+				const newMessages = action.payload.messages;
+				const taskSuggestionsMessagePayload: TaskMessage[] = [];
+
+				for (const message of newMessages) {
+					if ("text" in message.payload) {
+						state.messages.push(message);
+					}
+
+					if ("task" in message.payload) {
+						taskSuggestionsMessagePayload.push(message.payload);
+						state.taskSuggestions.push({
+							categoryId: message.payload.task.categoryId,
+							categoryName: message.payload.task.categoryName,
+							description: message.payload.task.description,
+							label: message.payload.task.label,
+						});
+					}
+				}
+
```

Comment:
I would suggest putting this kind of processing in the action, leaving slice only for state updates.

Reply:
And maybe it would be more convenient to use map instead of for?

Reply:
I tried to use map initially but i think that the linter preferred for ... of

Reply:
Agree with @anogire, the logic should be moved to action